pkg/scheduler/algorithmprovider/plugins_test.go

```diff
@@ -81,8 +78,8 @@ func TestApplyFeatureGates(t *testing.T) {
 				t.Fatalf("Error retrieving provider: %v", err)
 			}
 
-			if !p.FitPredicateKeys.Has("CheckNodeCondition") {
-				t.Fatalf("Failed to find predicate: 'CheckNodeCondition'")
+			if p.FitPredicateKeys.Has("CheckNodeCondition") {
```

Comment:
expected?

Reply:
it is fine, we could remove it completely since it is not guarded by feature gate anymore.